bundles/net.tourbook/src/net/tourbook/data/TourData.java

```diff
@@ -7233,44 +7258,58 @@ public int getConconiDeflection() {
     */
    public HashMap<String, float[]> getCustomTracks() {
 
-      if (_customTracks_UI.isEmpty()) {
-
-         if (customTracks != null && !customTracks.isEmpty()) {
-
-            // create UI data serie
-            for (final String idx : customTracks.keySet()) {
-               if (customTracks.get(idx) == null) {
-                  continue;
-               }
-               final int serieSize = customTracks.get(idx).length;
-               final float[] serieData = customTracks.get(idx);
-               final float[] serieValues = new float[serieSize];
-               for (int serieIndex = 0; serieIndex < serieSize; serieIndex++) {
-                  serieValues[serieIndex] = serieData[serieIndex];
-               }
-               _customTracks_UI.put(idx, serieValues);
+      if (_customTracks != null && _customTracks.size() > 0) {
+         if (_customTracks_UI == null) {
+            _customTracks_UI = new HashMap<>();
+         } else {
+            _customTracks_UI.clear();
+         }
+         // create UI data serie
+         for (final String idx : _customTracks.keySet()) {
+            if (_customTracks.get(idx) == null) {
+               continue;
+            }
+            final int serieSize = _customTracks.get(idx).length;
+            final float[] serieData = _customTracks.get(idx);
+            final float[] serieValues = new float[serieSize];
+            for (int serieIndex = 0; serieIndex < serieSize; serieIndex++) {
+               serieValues[serieIndex] = serieData[serieIndex];
             }
+            _customTracks_UI.put(idx, serieValues);
          }
       }
 
+      if (_customTracks_UI == null) {
+         return new HashMap();
+      }
+
+
       return _customTracks_UI;
    }
 
 
    //CUSTOM TRACKS
    public float[] getCustomTracks(final String idx) {
 
+      if (_customTracks == null) {
+         return null;
+      }
+      if (_customTracks_UI == null) {
+         _customTracks_UI= new HashMap();
+      }
```

Comment:
This is because , one of your eclipse settings remove the type for HashMap, so it seems to be a contradiction here or I missed something. when I put HashMap<K,V> when I save the "<K,V>" part is remove beccause of the eclipse setting!

Reply:
@wolfgang-ch , in you eclipse setting  "Java"->Editor->save action , you ask to put the setting "Remove redundant type arguments". which remove the argument of the hashmap when saving. 
So what is the solution you advice if you don't want the warning on missing parameters when having new HashMap ?

Reply:
before `Add type arguments to 'HashMap'`

![mt-before-1](https://user-images.githubusercontent.com/1283445/104843441-f317e680-58ca-11eb-9add-9129d30bd054.png)

after `Add type arguments to 'HashMap'`

![mt-before-2](https://user-images.githubusercontent.com/1283445/104843445-f6ab6d80-58ca-11eb-8cea-eef8b6ac5528.png)

after saving -> yellow flag is gone

![mt-after-save](https://user-images.githubusercontent.com/1283445/104843444-f6ab6d80-58ca-11eb-9522-fd36806d4ff2.png)

Reply:
@wolfgang-ch  in your **eclipse setting "Java"->Editor->save action , you ask to put the setting "Remove redundant type arguments"**. 
So yes it is the solution but your settings automatically remove it after I save, I need to press the "Ctrl+z" after saving the file to have the parameters back!
**So should I just uncheck in the  in eclipse setting "Java"->Editor->save action , the setting "Remove redundant type arguments"** ?

Reply:
I just checked my save actions with this in the docu https://mytourbook.sourceforge.io/mytourbook/images/mytb/dev/pref-settings-java-editor-save_actions-1903.png and both are the same.

 I don't need to press `<Ctrl>` + `Z` after saving to keep the `<>`.

Reply:
> I just checked my save actions with this in the docu https://mytourbook.sourceforge.io/mytourbook/images/mytb/dev/pref-settings-java-editor-save_actions-1903.png and both are the same.
> 
> I don't need to press `<Ctrl>` + `Z` after saving to keep the `<>`.

Me too I have the exact same setting as the one in the docu.
 but for me the very last setting "**Remove redundant type Arguments**" does its job and remove the '<>' after I save.
very strange.